### PR TITLE
Split collections to allow dropping Kubernetes import

### PR DESCRIPTION
**Pull request by howardjohn**
For https://github.com/istio/istio/issues/26232

This code is admittedly pretty bad - but gets the job done

Importing k8s libraries adds up to 40mb to the binary. This, along with many other changes, allows completely dropping the imports and cutting the agent binary size in half. See https://github.com/howardjohn/istio/pull/new/agent/tiny-binary-full-branch for this end state.

### CODE REVIEW

File: pkg/config/schema/codegen/tools/collections.main.go [hunk 1]
```diff
@@ -28,14 +29,20 @@ import (
...
+	if len(os.Args) != 4 && len(os.Args) != 7 {
 		fmt.Printf("Invalid args: %v", os.Args)
 		os.Exit(-1)
 	}
 
 	pkg := os.Args[1]
 	input := os.Args[2]
 	output := os.Args[3]
+	var splitOn, splitOutput, splitPrefix string
+	if len(os.Args) > 4 {
```
Comment (nmittler) [inline pkg/config/schema/codegen/tools/collections.main.go:41]: Is the intent to panic if there are only 4 or 5 elements?

Comment (howardjohn) [reply pkg/config/schema/codegen/tools/collections.main.go:41]: I think L32 should ensure its either exactly 4 or 7. This code has gotten pretty sketchy over time though (especially with this PR)

**Review by nmittler — APPROVED** [review]

File: pkg/config/schema/codegen/tools/collections.main.go (hunk 1 above)
Comment (nmittler) [reply pkg/config/schema/codegen/tools/collections.main.go:41]: ah right ... missed that. Maybe expand the error message a bit to explain the valid conditions more clearly?

**Review by hzxuzhonghu — APPROVED** [review]

File: galley/pkg/config/testing/basicmeta/generate.go [hunk 2]
```diff
@@ -21,3 +21,4 @@ package basicmeta
 //go:generate go run $REPO_ROOT/pkg/config/schema/codegen/tools/collections.main.go basicmeta basicmeta.yaml collections.gen.go
 
 //go:generate goimports -w -local istio.io "$REPO_ROOT/galley/pkg/config/testing/basicmeta/collections.gen.go"
+//go:generate goimports -w -local istio.io "$REPO_ROOT/galley/pkg/config/testing/basicmeta/collections.gen.go"
```
Comment (hzxuzhonghu) [inline galley/pkg/config/testing/basicmeta/generate.go:24]: isn't this duplicate

**Pull request merged**